epochStart/bootstrap/shardStorageHandler.go

```diff
@@ -140,86 +141,97 @@ func (ssh *shardStorageHandler) SaveDataToStorage(components *ComponentsNeededFo
 		return err
 	}
 
-	err = ssh.saveTries(components)
+	err = ssh.commitTries(components)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (ssh *shardStorageHandler) getProcessMiniBlocks(
-	pendingMiniBlocks map[string]*block.MiniBlock,
-	meta *block.MetaBlock,
-	headers map[string]data.HeaderHandler,
-) ([]bootstrapStorage.MiniBlocksInMeta, error) {
-	processedMiniBlocks := make([]bootstrapStorage.MiniBlocksInMeta, 0)
-	for _, epochStartShardData := range meta.EpochStart.LastFinalizedHeaders {
-		if epochStartShardData.ShardID != ssh.shardCoordinator.SelfId() {
+func getEpochStartShardData(metaBlock *block.MetaBlock, shardId uint32) (block.EpochStartShardData, error) {
+	for _, epochStartShardData := range metaBlock.EpochStart.LastFinalizedHeaders {
+		if epochStartShardData.ShardID != shardId {
 			continue
 		}
 
-		neededMeta, ok := headers[string(epochStartShardData.FirstPendingMetaBlock)].(*block.MetaBlock)
-		if !ok {
-			return nil, epochStart.ErrMissingHeader
-		}
+		return epochStartShardData, nil
+	}
 
-		processedMbHashes := make([][]byte, 0)
-		miniBlocksDstMe := getAllMiniBlocksWithDst(neededMeta, ssh.shardCoordinator.SelfId())
-		for hash, mb := range miniBlocksDstMe {
-			if _, ok := pendingMiniBlocks[hash]; ok {
-				continue
-			}
+	return block.EpochStartShardData{}, epochStart.ErrEpochStartDataForShardNotFound
+}
 
-			processedMbHashes = append(processedMbHashes, mb.Hash)
-		}
+func (ssh *shardStorageHandler) getProcessedMiniBlocks(
+	pendingMiniBlocks map[string]*block.MiniBlock,
+	meta *block.MetaBlock,
+	headers map[string]data.HeaderHandler,
+) ([]bootstrapStorage.MiniBlocksInMeta, error) {
+	shardData, err := getEpochStartShardData(meta, ssh.shardCoordinator.SelfId())
+	if err != nil {
+		return nil, err
+	}
 
-		processedMiniBlocks = append(processedMiniBlocks, bootstrapStorage.MiniBlocksInMeta{
-			MetaHash:         epochStartShardData.FirstPendingMetaBlock,
-			MiniBlocksHashes: processedMbHashes,
-		})
-		return processedMiniBlocks, nil
+	neededMeta, ok := headers[string(shardData.FirstPendingMetaBlock)].(*block.MetaBlock)
+	if !ok {
+		return nil, epochStart.ErrMissingHeader
```

Comment:
Perhaps the same error kind as below (not sure)?

Reply:
changed below